hpcgap/lib/oper.g

```diff
@@ -1276,6 +1276,106 @@ BIND_GLOBAL( "DeclareAttribute", function ( arg )
 end );
 
 
+##############################################################################
+##
+##  This code adds thread-local attributes, which are used in HPC-GAP.
+##  These are not complete replacements for normal attributes.
+##  In particular, they are not filters, so cannot be used in method selection.
+##
+##  The main problem with thread-local attributes is that only a subset of the
+##  threads may have a value for a particular mutable attribute, but filters
+##  are not stored thread-locally. Therefore thread-local attributes are not
+##  a perfect replacement for traditional attributes.
+##
+##  The reason we have thread-local attributes is for code which uses
+##  mutable attribute as a general scratch-space, like StabChainMutable, which
+##  cannot be shared between threads.
+##
+##  Thread-local attributes where introduced to make existing
+##  GAP mutable attributes safe to use in a multi-threaded way, without
+##  significant code changes. If possible, no new thread local attributes should
+##  be added, and existing ones removed and cleaned up where possible.
+##
+##  Current Limitations:
+##
+##  In HPCGAP we disable the ImagesSet overload in lib/ghomperm.gi as it filters
+##  on HasStabChainMutable
+##
+##
+##  Thread local attributes must be mutable. The mutflag option is kept
```

Comment:
Aha, so I guessed right above -- as I said there, I'd lead with this information!

Also, immutable TLS attributes don't make sense to me anyway, so I am not sure why this is listed under "limitations" anyway?